packages/gestalt/src/TableRowExpandable.js

```diff
@@ -0,0 +1,52 @@
+// @flow strict
+import React, { type Node, useState } from 'react';
+import cx from 'classnames';
+import styles from './Table.css';
+import Box from './Box.js';
+import IconButton from './IconButton.js';
+
+type Props = {|
+  accessibilityExpandLabel: string,
+  accessibilityCollapseLabel: string,
+  children: Node,
+  hoverStyle?: 'gray' | 'none',
+  expandedContents: Node,
+|};
+
+export default function TableRowExpandable(props: Props): Node {
+  const {
+    accessibilityCollapseLabel,
+    accessibilityExpandLabel,
+    children,
+    expandedContents,
+  } = props;
+  const [expanded, setExpanded] = useState(false);
+  const hoverStyle = props.hoverStyle || 'gray';
+  const cs = hoverStyle === 'gray' ? cx(styles.hoverShadeGray) : null;
+
+  return (
+    <>
+      <tr className={cs}>
+        <td>
+          <IconButton
+            accessibilityLabel={
+              expanded ? accessibilityCollapseLabel : accessibilityExpandLabel
+            }
+            icon={expanded ? 'arrow-up' : 'arrow-down'}
```

Comment:
@faisalgedi we also need `accessibilityExpanded` and `accessibilityControls` on this `IconButton`.

```
accessibilityExpanded={expanded}
accessibilityControls={id}
```

The id should then be specified on the expanded `tr`:

```
      {expanded && (
        <tr id={id}>
          <td colSpan={React.Children.count(children) + 1}>
            <Box padding={6}>{expandedContents}</Box>
          </td>
        </tr>
      )}
```

So we should add an `id` and make it required for every `Table.ExpandableRow`